pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaSSLUtils.java

```diff
@@ -0,0 +1,340 @@
+  }
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaSSLUtils.class);
+  // Value constants
+  private static final String DEFAULT_CERTIFICATE_TYPE = "X.509";
+  private static final String DEFAULT_KEY_ALGORITHM = "RSA";
+  private static final String DEFAULT_KEYSTORE_TYPE = "PKCS12";
+  private static final String DEFAULT_SECURITY_PROTOCOL = "SSL";
+  private static final String DEFAULT_TRUSTSTORE_TYPE = "jks";
+  private static final String DEFAULT_SERVER_ALIAS = "ServerAlias";
+  private static final String DEFAULT_CLIENT_ALIAS = "ClientAlias";
+  // Key constants
+  private static final String SSL_TRUSTSTORE_LOCATION = "ssl.truststore.location";
+  private static final String SSL_TRUSTSTORE_PASSWORD = "ssl.truststore.password";
+  private static final String SECURITY_PROTOCOL = "security.protocol";
+  private static final String SSL_KEYSTORE_LOCATION = "ssl.keystore.location";
+  private static final String SSL_KEYSTORE_PASSWORD = "ssl.keystore.password";
+  private static final String SSL_KEY_PASSWORD = "ssl.key.password";
+  private static final String STREAM_KAFKA_SSL_SERVER_CERTIFICATE = "stream.kafka.ssl.server.certificate";
+  private static final String STREAM_KAFKA_SSL_CERTIFICATE_TYPE = "stream.kafka.ssl.certificate.type";
+  private static final String SSL_TRUSTSTORE_TYPE = "ssl.truststore.type";
+  private static final String STREAM_KAFKA_SSL_CLIENT_CERTIFICATE = "stream.kafka.ssl.client.certificate";
+  private static final String STREAM_KAFKA_SSL_CLIENT_KEY = "stream.kafka.ssl.client.key";
+  private static final String STREAM_KAFKA_SSL_CLIENT_KEY_ALGORITHM = "stream.kafka.ssl.client.key.algorithm";
+  private static final String SSL_KEYSTORE_TYPE = "ssl.keystore.type";
+
+  public static void initSSL(Properties consumerProps) {
+    // Check if one-way SSL is enabled. In this scenario, the client validates the server certificate.
+    String trustStoreLocation = consumerProps.getProperty(SSL_TRUSTSTORE_LOCATION);
+    String trustStorePassword = consumerProps.getProperty(SSL_TRUSTSTORE_PASSWORD);
+    if (StringUtils.isAnyEmpty(trustStoreLocation, trustStorePassword)) {
+      return;
+    }
+    renewTrustStore(consumerProps);
+    initTrustStore(consumerProps);
+
+    // Set the security protocol
+    String securityProtocol = consumerProps.getProperty(SECURITY_PROTOCOL, DEFAULT_SECURITY_PROTOCOL);
+    consumerProps.setProperty(SECURITY_PROTOCOL, securityProtocol);
+
+    // Check if two-way SSL is enabled. In this scenario, the client validates the server's certificate and the server
+    // validates the client's certificate.
+    String keyStoreLocation = consumerProps.getProperty(SSL_KEYSTORE_LOCATION);
+    String keyStorePassword = consumerProps.getProperty(SSL_KEYSTORE_PASSWORD);
+    String keyPassword = consumerProps.getProperty(SSL_KEY_PASSWORD);
+    if (StringUtils.isAnyEmpty(keyStoreLocation, keyStorePassword, keyPassword)) {
+      return;
+    }
+    renewKeyStore(consumerProps);
```

Comment:
I think that it's probably better to represent the logic as:

```
if (shouldRenewKeyStore()) {
  initKeyStore(consumerProps);  // delete and recreate the key store file
}
```

Reply:
done.